scripts/lint_naming_conventions.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python
+
+# Script used to review jenkins jobs and report job names
+# not following conventions
+# https://github.com/rcbops/rpc-gating#naming-conventions
+
+import argparse
+import os
+import re
+import sys
+import yaml
+
+
+def parse_args():
+    description = "Parses JJB files to ensure naming conventions followed"
+    parser = argparse.ArgumentParser(description=description)
+    parser.add_argument("--exclude-files",
+                        required=False,
+                        help="Comma seperated list of files to exclude")
+    parser.add_argument("--dirs",
+                        required=True,
+                        help="Comma seperated dirs to parse")
+    parser.add_argument("--parse-jjb", required=False,
+                        help="Parse jjb file for names",
+                        action='store_true')
+    return parser.parse_args()
+
+
+def parse_jjb_file(in_dir, in_file):
+    _rc = 0
+    filename = os.path.join(in_dir, in_file)
+    file_yaml = yaml.safe_load(open(filename).read())
+    for item in file_yaml:
+        if "project" in item:
+            # check project name
+            if parse_job_name(item["project"]["name"], filename):
+                _rc = 1
+            # check project jobs names
+            for job in item["project"]["jobs"]:
+                if parse_job_name(job, filename):
+                    _rc = 1
+
+        if "job-template" in item:
+            # check job-template name
+            if parse_job_name(item["job-template"]["name"], filename):
+                _rc = 1
+
+        if "job" in item:
+            # check individual job
+            if parse_job_name(item["job"]["name"], filename):
+                _rc = 1
+    return _rc
+
+
+def parse_job_name(job_name, file_name):
+    regex = '^([a-zA-Z0-9]+-?)+([_]([{][a-zA-Z0-9]+[}]-?)+)?$'
+    match = re.match(regex, job_name)
+    if not match:
+        out = "Lint error: Job name \"%s\" in \"%s\"" % (job_name, file_name)
+        out += "\n\tdoes not conform to rpc-gating naming conventions.\n"
```

Comment:
I considered finding a good place to throw that out, and could be done at the end if there is a failure. But I thought otherwise because
* other lints don't provide links to the exact conditions being checked
* the naming standards are in the short readme of the rpc-gating repo, not buried in documentation or requiring a doc build process
* `lint.sh` output is already 227 lines

But, if adding ~2 lines of code saves someone 5 minutes or more in finding or asking for help in finding the naming conventions, seems worth it. I'll add it.